src/spec.c

```diff
@@ -2917,7 +2912,11 @@ int IndexSpec_CreateFromRdb(RedisModuleCtx *ctx, RedisModuleIO *rdb, int encver,
 
   sp->numFields = LoadUnsigned_IOError(rdb, goto cleanup);
   sp->fields = rm_calloc(sp->numFields, sizeof(FieldSpec));
-  int maxSortIdx = -1;
+  // First, initialise fields IndexError before loading them.
+  // If some fields are not loaded correctly, we will attempt to free the spec and cleanup the fields.
+  for (int i = 0; i < sp->numFields; i++) {
+    sp->fields[i].indexError = IndexError_Init();
+  }
```

Comment:
Will that fix the leak in `test_rdb_memory_limit`? Can we remove the skip for sanitizer there to validate (or test it in some other way)?

Reply:
Its not flaky, it's new from my PR from yesterday, and this is probably the only test that checks rdb failure with sanitizer...